java/src/test/android/app/src/androidTest/java/ai/onnxruntime/example/javavalidater/SimpleTest.java

```diff
@@ -0,0 +1,74 @@
+package ai.onnxruntime.example.javavalidater;
+
+import ai.onnxruntime.*;
+import ai.onnxruntime.OnnxTensor;
+import ai.onnxruntime.OrtSession;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import android.content.res.AssetManager;
+import android.util.Log;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.platform.app.InstrumentationRegistry;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import java.util.*;
+
+
+@RunWith(AndroidJUnit4.class)
+public class SimpleTest {
+
+    @Test
+    public void runSigmoidModelTest() throws OrtException, IOException {
+
+        try (OrtEnvironment env = OrtEnvironment.getEnvironment();
+             OrtSession.SessionOptions opts = new OrtSession.SessionOptions()) {
+
+            opts.setOptimizationLevel(OrtSession.SessionOptions.OptLevel.BASIC_OPT);
+
+            try (OrtSession session = env.createSession(readModel("sigmoid.ort"), opts)) {
+                String inputName = session.getInputNames().iterator().next();
+                float[][][] testdata = new float[3][4][5];
+                for (float[][] array2d : testdata) {
+                    for (float[] array : array2d) {
+                        Arrays.fill(array, (float) 0);
+                    }
+                }
+                try (OnnxTensor inputTensor =
+                             OnnxTensor.createTensor(env, testdata);
+                     OrtSession.Result output = session.run(Collections.singletonMap(inputName, inputTensor))) {
+                    float[][][] rawOutput = (float[][][]) output.get(0).getValue();
+                    //expected sigmoid output is y = 1.0 / (1.0 + exp(-x))
+                    float expected = (float) 0.5;
+                    for (float[][] array2d : rawOutput) {
+                        for (float[] array : array2d) {
+                            for (float actual : array) {
+                                Assert.assertEquals(actual, expected, 1e-15);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+
+    private byte[] readModel(String fileName) throws IOException {
+        AssetManager assetManager =
```

Comment:
Do we need to go via this manual process or can we just create the session by passing in the filename (as ORT has code to handle reading bytes from a file)?

Reply:
I tried to directly pass a model path into `ort.createsession()`, seems not working. (I am just guessing) is it because we are loading an ort format model but not an onnx format model?

Reply:
it should support ORT format models too. what kind of error do you see?

Reply:
> it should support ORT format models too. what kind of error do you see?

Sorry just saw this. The error message is `ai.onnxruntime.OrtException: Error code - ORT_INVALID_ARGUMENT - message: Invalid fd was supplied: -1` and I passed in a model path like `..\..\assets\sigmoid.ort`

Reply:
I see - perhaps the path is not valid. looking around online a bit, it seems that AssetManager is the way to access assets. looks like they are stored in some other format (https://gamedev.stackexchange.com/questions/165071/how-to-get-asset-folder-path-in-android-native-activity)

Reply:
The only path you can have read access is from '/data/local/tmp' (we use that for our UT), which is not recommended for actual Android App.